src/http_server.cc

```diff
@@ -3225,6 +3225,42 @@ HTTPAPIServer::HandleGenerate(
       req, RestrictedCategory::INFERENCE, restricted_apis_);
 
   AddContentTypeHeader(req, "application/json");
+#ifdef TRITON_ENABLE_METRICS
+  // logic to add kv_cache metrics to response header
+  // Get the metrics in Prometheus format
+  if (std::getenv("ORCA_HEADER_METRIC_TYPE") != nullptr) {
+    const std::string orca_type = std::getenv("ORCA_HEADER_METRIC_TYPE");
+    TRITONSERVER_Metrics* metrics = nullptr;
+    TRITONSERVER_Error* err =
+        TRITONSERVER_ServerMetrics(server_.get(), &metrics);
+    if (err == nullptr) {
+      const char* base;
+      size_t byte_size;
+      err = TRITONSERVER_MetricsFormatted(
+          metrics, TRITONSERVER_METRIC_PROMETHEUS, &base, &byte_size);
+      if (err == nullptr) {
+        std::string kv_utilization(base, byte_size);
+        // Extract the KV utilization metrics from the Prometheus formatted
+        // string.
+        std::string extracted_kv_metrics =
+            ExtractKVMetrics(kv_utilization, orca_type);
+        if (!extracted_kv_metrics.empty()) {
+          evhtp_headers_add_header(
+              req->headers_out,
+              evhtp_header_new(
+                  "endpoint-load-metrics", extracted_kv_metrics.c_str(), 1, 1));
+        }
+      }
+    }
+    TRITONSERVER_MetricsDelete(metrics);
```

Comment:
why deleting the metric?

Reply:
The [TRITONSERVER_ServerMetrics()](https://github.com/triton-inference-server/core/blob/main/src/tritonserver.cc#L3256) call returns a [TritonServerMetrics](https://github.com/triton-inference-server/core/blob/main/src/tritonserver.cc#L166) object that we use to get the serialized metrics. The [caller is responsible for calling delete](https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonserver.h#L2577) on it once they're done using it; it [doesn't actually delete the metric](https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonserver.h#L656) just the metrics object.